optimum/intel/openvino/modeling_decoder.py

```diff
@@ -950,6 +947,47 @@ def _from_pretrained(
         return causal_model
 
 
+class OVPhi3ForCausalLM(OVModelForCausalLM):
+    def prepare_inputs_for_generation(
+        self,
+        input_ids,
+        past_key_values=None,
+        attention_mask=None,
+        inputs_embeds=None,
+        cache_position=None,
+        position_ids=None,
+        use_cache=True,
+        logits_to_keep=None,
+        **kwargs,
+    ):
+        # Overwritten -- this model may need to switch between short and long rope, invalidating the cache in the
```

Comment:
Am I correct that we have a problem when we have short and long prompts in consecutive `generate` calls?
We can't re-initialize `inv_freqs` from `long_inv_freqs` to `short_inv_freqs` and vise-versa? How this problem is solved?

Reply:
Wouldn't this: https://github.com/huggingface/optimum-intel/pull/1504/files#r2518867960 solve the issue?

Reply:
As discussed offline: this is handled in optimum-intel by resetting the kv-cache when the number of input tokens is equal to the long rope boundary (e.g. 4096). This is done the same way in transformers code. Tested that this works as expected in chat context with https://gist.github.com/helena-intel/b55522cda91d9d61a644f153e71f0f98 .